src/mlpack/methods/logistic_regression/logistic_regression.hpp

```diff
@@ -98,9 +99,9 @@ class LogisticRegression
    *
    * @param optimizer Instantiated optimizer with instantiated error function.
    */
-  template<template<typename> class OptimizerType>
+  template<template<typename, typename ... PolicyTypeArg> class OptimizerType, typename ... PolicyTypeArg>
```

Comment:
I think you can make the line shorter by giving no name to the first `PolicyTypeArg`:

```
template<template<typename, typename...> class OptimizerType, typename... OptimizerTypeArgs>
```

You can also remove the other overload of `Train(OptimizerType&)`, this will work in its place.